lib/spack/spack/relocate.py

```diff
@@ -497,6 +497,7 @@ def _replace_prefix_bin(filename, byte_prefixes):
             # We only care about this problem if we are about to replace
             length_compatible = len(new_bytes) <= len(orig_bytes)
             if not length_compatible:
+                print('Failing filename', filename)
                 raise BinaryTextReplaceError(orig_bytes, new_bytes)
```

Comment:
currently trying to figure out how this occurred:
```
Failing filename /projects/wind/spack-manager/views/snapshots/._20210910/pxq6nab47hwwoiqbrc3j5ktotjuddytg/lib/libnetcdf.so.18.0.0
==> Error: New path longer than old path: binary text replacement not possible.
The new path b'/projects/wind/spack-manager/views/snapshots/._20210910/pxq6nab47hwwoiqbrc3j5ktotjuddytg' is longer than the old path b'/apps/mpi/4.0.3-ucx'.
```

Reply:
Ah figured it out.  the traversal of the spec is also capturing externals like `/usr` and adds them to the list of `prefix_to_projection`. I don't think this traversal is necessary. for copy we just want to move binary from location `a` to `b` we don't need to check any other locations for the same binary right?